### PR TITLE
libzmq: Support for no encryption in conan v2

### PR DESCRIPTION
**Pull request by gummif**
Specify library name and version:  **libzmq/all**

fixes https://github.com/conan-io/conan-center-index/issues/17289

---

- [ ] I've read the [contributing guidelines](https://github.com/conan-io/conan-center-index/blob/master/CONTRIBUTING.md).
- [ ] I've used a [recent](https://github.com/conan-io/conan/releases/latest) Conan client version close to the [currently deployed](https://github.com/conan-io/conan-center-index/blob/master/.c3i/config_v1.yml#L6).
- [ ] I've tried at least one configuration locally with the [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

File: recipes/zeromq/all/conanfile.py [hunk 1]
```diff
@@ -22,7 +22,7 @@ class ZeroMQConan(ConanFile):
     options = {
         "shared": [True, False],
         "fPIC": [True, False],
-        "encryption": [None, "libsodium", "tweetnacl"],
+        "encryption": [None, "none", "libsodium", "tweetnacl"], # None for conan v1 compat, use "none" in conan v2 for no encryption
```
Comment (prince-chrismc) [inline recipes/zeromq/all/conanfile.py:25]: See https://github.com/conan-io/conan-center-index/issues/17289#issuecomment-1526500000 for my rational :)

```suggestion
        "encryption": [False, "libsodium", "tweetnacl"],
```

File: recipes/zeromq/all/conanfile.py [hunk 2]
```diff
@@ -71,7 +71,7 @@ def source(self):
 
     def generate(self):
         tc = CMakeToolchain(self)
-        tc.variables["ENABLE_CURVE"] = bool(self.options.encryption)
+        tc.variables["ENABLE_CURVE"] = self.options.encryption == "libsodium" or self.options.encryption == "tweetnacl"
```
Comment (prince-chrismc) [inline recipes/zeromq/all/conanfile.py:74]:
```suggestion
        tc.variables["ENABLE_CURVE"] = bool(self.options.encryption)
```

**Review by prince-chrismc — COMMENTED** [review]
I think `False` is the best solution here, the `None` was a bit erroneous and I considered leaving the old value and having a warning saying it was deprecated but I would rather prioritize it being correct for 2.0 so this will force people to update and not have a problem migrating 🤞

File: recipes/zeromq/all/conanfile.py (hunk 1 above)
Comment (gummif) [reply recipes/zeromq/all/conanfile.py:25]: This would break v1 I guess, why not `"encryption": [None, False, "libsodium", "tweetnacl"],` ?

Comment (uilianries) [reply recipes/zeromq/all/conanfile.py:25]: Break how? Boolean of None is False. Please, let's keep it simpler, only False is enough for this case.

**Review by uilianries — APPROVED** [review]
LGTM

**Pull request merged**